### PR TITLE
tbb: update to 2021.11.0.

### PR DESCRIPTION
**Pull request by Calandracas606**
#### Testing the changes
- I tested the changes in this PR: **briefly**
- Tested TBB, but have not tested the bumped packages

#### Local build testing
- I built this PR locally for my native architecture, x86_64-musl
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - x86_64
  - i686
  - aarch64-musl (cross)
  - armv7l (cross)
  - armv6l-musl (cross)

Tested builds for bumped packages in all the above architectures (when supported)

Fixed the build for PrusaSlicer, however it segfaults immediately: https://github.com/void-linux/void-packages/issues/46687

[ci skip]

### CODE REVIEW

**Comment by Calandracas606** [comment]
@thypon FYI this is an update to a package for which you are listed as the maintainer

**Comment by thypon** [comment]
sounds good to me

File: srcpkgs/tbb/template [hunk 1]
```diff
@@ -1,76 +1,43 @@
...
-
 tbb-devel_package() {
 	depends="${sourcepkg}>=${version}_${revision}"
 	short_desc+=" - development files"
 	pkg_install() {
 		vmove usr/include
-		vmove "usr/lib/*.so"
 		vmove usr/lib/cmake
+		vmove usr/lib/pkgconfig
+		vmove "usr/lib/*.so"
```
Comment (cinerea0) [inline srcpkgs/tbb/template:41]:
```suggestion
		vmove "usr/lib/*.so"
```
You don't need to move this to the end.

File: srcpkgs/tbb/template [hunk 2]
```diff
@@ -1,76 +1,43 @@
...
-hostmakedepends="cmake"
+version=2021.11.0
+revision=1
+build_style=cmake
+configure_args="-DTBB_STRICT=OFF -DTBB_TEST=OFF"
+makedepends="libgomp-devel libhwloc-devel"
 short_desc="Intel Threading Building Blocks"
 maintainer="Andrea Brancaleoni <abc@pompel.me>"
 license="Apache-2.0"
 homepage="https://www.threadingbuildingblocks.org"
```
Comment (cinerea0) [inline srcpkgs/tbb/template:11]:
```suggestion
homepage="https://oneapi-src.github.io/oneTBB/"
```

File: srcpkgs/tbb/template [hunk 3]
```diff
@@ -1,76 +1,43 @@
...
+		makedepends+=" python3-devel"
+	;;
 esac
 
 post_extract() {
-	sed -i 's|CPLUS = g..|CPLUS = $(CXX)|g' build/linux.gcc.inc
-	sed -i 's|CONLY = gcc|CONLY = $(CC)|g' build/linux.gcc.inc
-
 	# alternative might be:
 	# https://git.alpinelinux.org/cgit/aports/tree/testing/libtbb/glibc-struct-mallinfo.patch
```
Comment (cinerea0) [inline srcpkgs/tbb/template:27]: These comments haven't been relevant for a while, you can remove them.

File: srcpkgs/tbb/template [hunk 4]
```diff
@@ -1,76 +1,43 @@
...
-	*)
-		# make sure to fall back instead of letting host uname control it
-		make_build_args="arch=generic"
-		;;
+	x86_64*|aarch64*)
+		build_options_default+=" -DTBB4PY_BUILD=ON"
+		hostmakedepends+=" python3-devel python3-distutils-extra swig"
+		makedepends+=" python3-devel"
+	;;
 esac
```
Comment (cinerea0) [inline srcpkgs/tbb/template:23]:
```suggestion
esac

if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
	makedepends+=" libucontext-devel"
	configure_args+=" -DCMAKE_CXX_STANDARD_LIBRARIES=-lucontext"
fi

if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
	makedepends+=" libatomic-devel"
	configure_args+=" -DCMAKE_CXX_FLAGS=-latomic"
fi
```

I found these build options fixed some problems. The `libucontext` one in particular enables relocation on musl.

**Review by cinerea0 — COMMENTED** [review]
Suggesting some changes from my open PR (#48733). I'm okay with closing that one in favor of this one if you've done significant building/testing. Building the dependent packages should let you know if there are more changes you need to make to `common/shlibs`.

File: srcpkgs/tbb/template (hunk 4 above)
Comment (Calandracas606) [reply srcpkgs/tbb/template:23]: I've been added these in. I haven't encountered any issues with musl without libucontext, and it doesn't require the shlib either

**Comment by Calandracas606** [comment]
> Suggesting some changes from my open PR (#48733). I'm okay with closing that one in favor of this one if you've done significant building/testing. Building the dependent packages should let you know if there are more changes you need to make to `common/shlibs`.

Sorry I should have checked before opening this PR

As aforementioned, I have done extensive build tests of the dependencies, and have been using this library for the past couple of days without issues

**Comment by thypon** [comment]
Any other issue here?

**Comment by cinerea0** [comment]
I don't think so.

File: srcpkgs/tbb/template [hunk 5]
```diff
@@ -1,76 +1,52 @@
...
-		;;
-	ppc*)
-		make_build_args="arch=ppc32"
-		;;
-	*)
-		# make sure to fall back instead of letting host uname control it
-		make_build_args="arch=generic"
-		;;
+	x86_64*|aarch64*)
+		build_options_default+=" -DTBB4PY_BUILD=ON"
```
Comment (sgn) [inline srcpkgs/tbb/template:19]: This is broken https://github.com/void-linux/void-packages/pull/48993

Comment (Calandracas606) [reply srcpkgs/tbb/template:19]: yes this is a mistake

**Pull request merged**